### ISSUE TITLE
[Feature request] python 3.10 supports

### ISSUE THREAD

**Issue by noraj** [dolevf/Damn-Vulnerable-GraphQL-Application#66]
I have seen:

- #34
- the readme saying `Note: Python 3.10 is not supported yet!`
- the [Dockerfile](https://github.com/dolevf/Damn-Vulnerable-GraphQL-Application/blob/207a71cbbda873c5c5ca5ff7fd4934877b5d17dc/Dockerfile#L1) using python 3.7

Would be nice to have python 3.10 support for server install.

**Comment by dolevf** [comment]
Yeah, so I looked into this and it appears the websockets/graphql subscription functionality is broken and not currently supported in 3.10. We will need to swap out flask-sockets library with something which is not a small lift. We may look into this in the future but at this point we will put this on ice.

**Comment by noraj** [comment]
# Subscription request

With Python 3.10 from the `dvga-py3-10`

![image](https://user-images.githubusercontent.com/16578570/208527856-b28ac70a-7368-4b18-837e-7dcd0a4ceb53.png)

With Python 3.7 from the dockerhub image

![image](https://user-images.githubusercontent.com/16578570/208529222-638b7967-c269-4a1a-a867-3318a97de498.png)

So I don't have a working subscription request that I could test.

**Comment by dolevf** [comment]
you can use Altair Client, I believe GraphiQL doesn't support subscriptions

**Comment by noraj** [comment]
> you can use Altair Client, I believe GraphiQL doesn't support subscriptions

I tried it with Insomnia too, that claims to support websockets https://github.com/Kong/insomnia/discussions/4477, cf. https://insomnia.rest/changelog#2022.6.0 and https://docs.insomnia.rest/insomnia/websockets. So [I asked if it supports GraphQL subscriptions](https://github.com/Kong/insomnia/discussions/5561).

Well then, I'll try again with Altair.

**Comment by dolevf** [comment]
I'm going to look into the errors that you're observing. Another way you can explore this is:
1. Load the UI
2. Go to Public Pastes
3. Create a paste with the mutation createPaste
4. Observe that the Public Paste page (step #2) updated the page immediately without a refresh

**Comment by dolevf** [comment]
https://user-images.githubusercontent.com/5857304/208740937-cf0c970f-edfe-495e-9b9e-acdaf94ea159.mp4

**Comment by noraj** [comment]
Subscriptions seems to work well on my side.

**Comment by noraj** [comment]
While we are at it we may update it to [Python 3.11](https://docs.python.org/3/whatsnew/3.11.html).

**Comment by dolevf** [comment]
hmm, that will require some additional work, I'll try to find some time.
